staging/src/k8s.io/client-go/rest/fake/fake.go

```diff
@@ -94,14 +96,20 @@ func (c *RESTClient) request(verb string) *restclient.Request {
 
 	ns := c.NegotiatedSerializer
 	info, _ := runtime.SerializerInfoForMediaType(ns.SupportedMediaTypes(), runtime.ContentTypeJSON)
-	internalVersion := schema.GroupVersion{
```

Comment:
The original code is consistent with the real rest client: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/rest/client.go#L189-L190, which may or may not be the proper way to write the code, but I prefer to keep the fake code align with the real code as much as possible.

The clientset uses the DirectCodec to avoid conversion: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/kubernetes/typed/core/v1/core_client.go#L147. Can you use the same DirectCodecFactory in your test?